packages/next-on-pages/src/buildApplication/generateFunctionsMap.ts

```diff
@@ -349,6 +368,104 @@ async function extractAndFixWasmRequires(
 	return { wasmIdentifiers, updatedContents };
 }
 
+/**
+ * Given the path of a function file and its content update the content so that it doesn't include Next.js manifests but imports them
+ * from the __next-on-pages-dist__/nextjs-manifests directory.
+ *
+ * Note: Such manifests are always the same for all functions so we need to share them across the various function files instead of costly duplicating them in each file.
+ *
+ * As a side effect this function also updates the file with the new content.
```

Comment:
Why does this function write to the disk while other functions that transform the content do not?
It feels a bit awkward that at the top level it is not obvious when the file on disk gets mutated.

Reply:
I did it here to be consistent with this function:
https://github.com/cloudflare/next-on-pages/blob/c0ab05a460ca430a73495f14a39a6cda5c505e1b/packages/next-on-pages/src/buildApplication/generateFunctionsMap.ts#L308

this might actually be wrong/unnecessary in both functions, I'll look into this separately (if that's ok 🙇 🙏 )